x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeRegistry.java

```diff
@@ -35,11 +35,8 @@ public Collection<DataType> dataTypes() {
 
     @Override
     public DataType fromEs(String typeName, TimeSeriesParams.MetricType metricType) {
-        if (metricType == TimeSeriesParams.MetricType.COUNTER) {
-            return EsqlDataTypes.getCounterType(typeName);
-        } else {
-            return EsqlDataTypes.fromName(typeName);
-        }
+        DataType type = EsqlDataTypes.fromName(typeName);
+        return metricType == TimeSeriesParams.MetricType.COUNTER ? type.widenSmallNumeric().counter() : type;
```

Comment:
It's not obvious to me what this is doing, or why it's equivalent to the old code.  Please add a comment.

Reply:
I added this to #110105.